api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -219,16 +220,30 @@ public static TimestampType withoutZone() {
       return INSTANCE_WITHOUT_ZONE;
     }
 
+    /**
+     * @return Timestamp type (with timezone) represented as INT96. This is only added for compatibility reasons
+     * and can only be written using a Spark's ParquetWriteSupport. Writing this type should be avoided.
+     */
```

Comment:
I don't think we should change the type system to support this. INT96 may be something that we can read, but Iceberg cannot write it, per the spec.

Was this needed to build the tests?

Reply:
Agreed. I found a way to have tests running that doesn't add a new type, I had to create an implementation of `ParquetWriter.Builder` that uses Spark's `ParquetWriteSupport` and Iceberg's `ParquetWriteAdapter` to avoid creating a `SparkSession`.